uboot-imx: bump to 2018.03 which fixes the build issues with fdt64_t redefinitions

v-vid opened the pull request:
* change mx6qsabresd to mx6qsabres to match defconfig name
* merge wanboard profiles since there is only one defconfig for target device
* remove unnecessary patches

Signed-off-by: Vladimir Vid <vladimir.vid@sartura.hr>

Commit:
uboot-imx: bump to 2018.03 which fixes the build issues with fdt64_t redefinitions

* change mx6qsabresd to mx6qsabres to match defconfig name
* merge wanboard profiles since there is only one defconfig for the target device
* move wanboard options from wandboard.h to defconfig
* remove legacy patches

Signed-off-by: Vladimir Vid <vladimir.vid@sartura.hr>

lperkov commented:
Ack from me.

blogic commented:
the patches are unnecessary as they are already upstream ?

v-vid commented:
All changes contained in the removed patches are mainlined except these two:

* 102-compiler_gcc-prevent-redefining-attributes.patch - removes some warning messages for the FreeBSD during the compile time.
* 110-wandboard-owrt-env.patch - conflicts with the updated wandboard.h and we should target mainline approach for booting the device

The following were breaking during compile and were removed since they don't actually affect any functionality. Build for all u-boot targets was successfully tested afterwards.

tmn505 commented:
Works fine on CuBoxTV (I4P TV-300-D).

hauke commented:
This does not compile for me I ma getting this error message:
```
In file included from include/common.h:48:0,
                 from common/image-fit.c:18:
common/image-fit.c: In function 'fit_get_debug':
common/image-fit.c:115:40: warning: implicit declaration of function 'fit_get_name'; did you mean 'eth_get_name'? [-Wimplicit-function-declaration]
        prop_name, (ulong)fit, noffset, fit_get_name(fit, noffset, NULL),
                                        ^
include/log.h:135:26: note: in definition of macro 'debug_cond'
    printf(pr_fmt(fmt), ##args); \
                          ^~~~
common/image-fit.c:114:2: note: in expansion of macro 'debug'
  debug("Can't get '%s' property from FIT 0x%08lx, node: offset %d, name %s (%s)\n",
  ^~~~~
common/image-fit.c:116:8: warning: implicit declaration of function 'fdt_strerror'; did you mean 'flash_perror'? [-Wimplicit-function-declaration]
        fdt_strerror(err));
        ^
include/log.h:135:26: note: in definition of macro 'debug_cond'
    printf(pr_fmt(fmt), ##args); \
                          ^~~~
common/image-fit.c:114:2: note: in expansion of macro 'debug'
  debug("Can't get '%s' property from FIT 0x%08lx, node: offset %d, name %s (%s)\n",
  ^~~~~
In file included from include/linux/bug.h:7:0,
                 from include/common.h:26,
                 from common/image-fit.c:18:
common/image-fit.c:114:8: warning: format '%s' expects argument of type 'char *', but argument 5 has type 'int' [-Wformat=]
  debug("Can't get '%s' property from FIT 0x%08lx, node: offset %d, name %s (%s)\n",
        ^
include/linux/printk.h:37:21: note: in definition of macro 'pr_fmt'
 #define pr_fmt(fmt) fmt
                     ^~~
include/log.h:142:2: note: in expansion of macro 'debug_cond'
  debug_cond(_DEBUG, fmt, ##args)
  ^~~~~~~~~~
common/image-fit.c:114:2: note: in expansion of macro 'debug'
  debug("Can't get '%s' property from FIT 0x%08lx, node: offset %d, name %s (%s)\n",
  ^~~~~
common/image-fit.c:114:8: warning: format '%s' expects argument of type 'char *', but argument 6 has type 'int' [-Wformat=]
  debug("Can't get '%s' property from FIT 0x%08lx, node: offset %d, name %s (%s)\n",
        ^
include/linux/printk.h:37:21: note: in definition of macro 'pr_fmt'
 #define pr_fmt(fmt) fmt
                     ^~~
include/log.h:142:2: note: in expansion of macro 'debug_cond'
  debug_cond(_DEBUG, fmt, ##args)
  ^~~~~~~~~~
common/image-fit.c:114:2: note: in expansion of macro 'debug'
  debug("Can't get '%s' property from FIT 0x%08lx, node: offset %d, name %s (%s)\n",
  ^~~~~
common/image-fit.c: In function 'fit_get_subimage_count':
common/image-fit.c:135:13: warning: implicit declaration of function 'fdt_next_node' [-Wimplicit-function-declaration]
   noffset = fdt_next_node(fit, images_noffset, &ndepth);
             ^~~~~~~~~~~~~
common/image-fit.c: In function 'fit_print_contents':
common/image-fit.c:176:8: warning: implicit declaration of function 'fit_get_desc'; did you mean 'fit_get_debug'? [-Wimplicit-function-declaration]
  ret = fit_get_desc(fit, 0, &desc);
        ^~~~~~~~~~~~
        fit_get_debug
common/image-fit.c:184:9: warning: implicit declaration of function 'fit_get_timestamp'; did you mean 'fit_get_debug'? [-Wimplicit-function-declaration]
   ret = fit_get_timestamp(fit, 0, &timestamp);
         ^~~~~~~~~~~~~~~~~
         fit_get_debug
common/image-fit.c:193:19: warning: implicit declaration of function 'fdt_path_offset' [-Wimplicit-function-declaration]
  images_noffset = fdt_path_offset(fit, FIT_IMAGES_PATH);
                   ^~~~~~~~~~~~~~~
common/image-fit.c:195:48: warning: format '%s' expects argument of type 'char *', but argument 3 has type 'int' [-Wformat=]
   printf("Can't find images parent node '%s' (%s)\n",
                                               ~^
                                               %d
          FIT_IMAGES_PATH, fdt_strerror(images_noffset));
                           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
common/image-fit.c:210:26: warning: format '%s' expects argument of type 'char *', but argument 4 has type 'int' [-Wformat=]
    printf("%s Image %u (%s)\n", p, count++,
                         ~^
                         %d
           fit_get_name(fit, noffset, NULL));
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
common/image-fit.c:213:4: warning: implicit declaration of function 'fit_image_print'; did you mean 'fit_image_load'? [-Wimplicit-function-declaration]
    fit_image_print(fit, noffset, p);
    ^~~~~~~~~~~~~~~
    fit_image_load
In file included from include/linux/bug.h:7:0,
                 from include/common.h:26,
                 from common/image-fit.c:18:
common/image-fit.c:220:9: warning: format '%s' expects argument of type 'char *', but argument 3 has type 'int' [-Wformat=]
   debug("Can't get configurations parent node '%s' (%s)\n",
         ^
include/linux/printk.h:37:21: note: in definition of macro 'pr_fmt'
 #define pr_fmt(fmt) fmt
                     ^~~
include/log.h:142:2: note: in expansion of macro 'debug_cond'
  debug_cond(_DEBUG, fmt, ##args)
  ^~~~~~~~~~
common/image-fit.c:220:3: note: in expansion of macro 'debug'
   debug("Can't get configurations parent node '%s' (%s)\n",
   ^~~~~
common/image-fit.c:226:18: warning: implicit declaration of function 'fdt_getprop'; did you mean 'fdt_getprop_u32'? [-Wimplicit-function-declaration]
  uname = (char *)fdt_getprop(fit, noffset, FIT_DEFAULT_PROP, NULL);
                  ^~~~~~~~~~~
                  fdt_getprop_u32
common/image-fit.c:240:34: warning: format '%s' expects argument of type 'char *', but argument 4 has type 'int' [-Wformat=]
    printf("%s Configuration %u (%s)\n", p, count++,
                                 ~^
                                 %d
           fit_get_name(fit, noffset, NULL));
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
common/image-fit.c:243:4: warning: implicit declaration of function 'fit_conf_print' [-Wimplicit-function-declaration]
    fit_conf_print(fit, noffset, p);
    ^~~~~~~~~~~~~~
In file included from include/linux/bug.h:7:0,
                 from include/common.h:26,
                 from common/image-fit.c:18:
common/image-fit.c: In function 'fit_image_print_data':
common/image-fit.c:273:8: warning: format '%s' expects argument of type 'char *', but argument 4 has type 'int' [-Wformat=]
  debug("%s  %s node:    '%s'\n", p, type,
        ^
include/linux/printk.h:37:21: note: in definition of macro 'pr_fmt'
 #define pr_fmt(fmt) fmt
                     ^~~
include/log.h:142:2: note: in expansion of macro 'debug_cond'
  debug_cond(_DEBUG, fmt, ##args)
  ^~~~~~~~~~
common/image-fit.c:273:2: note: in expansion of macro 'debug'
  debug("%s  %s node:    '%s'\n", p, type,
  ^~~~~
common/image-fit.c:276:6: warning: implicit declaration of function 'fit_image_hash_get_algo'; did you mean 'fit_image_print_data'? [-Wimplicit-function-declaration]
  if (fit_image_hash_get_algo(fit, noffset, &algo)) {
      ^~~~~~~~~~~~~~~~~~~~~~~
      fit_image_print_data
common/image-fit.c:281:10: warning: assignment makes pointer from integer without a cast [-Wint-conversion]
  keyname = fdt_getprop(fit, noffset, "key-name-hint", NULL);
          ^
common/image-fit.c:282:57: warning: comparison between pointer and integer
  required = fdt_getprop(fit, noffset, "required", NULL) != NULL;
                                                         ^~
common/image-fit.c:289:8: warning: implicit declaration of function 'fit_image_hash_get_value'; did you mean 'fit_image_print_data'? [-Wimplicit-function-declaration]
  ret = fit_image_hash_get_value(fit, noffset, &value,
        ^~~~~~~~~~~~~~~~~~~~~~~~
        fit_image_print_data
common/image-fit.c: In function 'fit_image_print_verification_data':
common/image-fit.c:335:7: warning: assignment makes pointer from integer without a cast [-Wint-conversion]
  name = fit_get_name(fit, noffset, NULL);
       ^
common/image-fit.c: At top level:
common/image-fit.c:359:6: warning: conflicting types for 'fit_image_print'
 void fit_image_print(const void *fit, int image_noffset, const char *p)
      ^~~~~~~~~~~~~~~
common/image-fit.c:213:4: note: previous implicit declaration of 'fit_image_print' was here
    fit_image_print(fit, noffset, p);
    ^~~~~~~~~~~~~~~
common/image-fit.c: In function 'fit_image_print':
common/image-fit.c:389:2: warning: implicit declaration of function 'fit_image_get_type'; did you mean 'image_get_type'? [-Wimplicit-function-declaration]
  fit_image_get_type(fit, image_noffset, &type);
  ^~~~~~~~~~~~~~~~~~
  image_get_type
common/image-fit.c:392:2: warning: implicit declaration of function 'fit_image_get_comp'; did you mean 'image_get_comp'? [-Wimplicit-function-declaration]
  fit_image_get_comp(fit, image_noffset, &comp);
  ^~~~~~~~~~~~~~~~~~
  image_get_comp
common/image-fit.c:395:8: warning: implicit declaration of function 'fit_image_get_data'; did you mean 'fit_image_print_data'? [-Wimplicit-function-declaration]
  ret = fit_image_get_data(fit, image_noffset, &data, &size);
        ^~~~~~~~~~~~~~~~~~
        fit_image_print_data
common/image-fit.c:418:3: warning: implicit declaration of function 'fit_image_get_arch'; did you mean 'image_get_arch'? [-Wimplicit-function-declaration]
   fit_image_get_arch(fit, image_noffset, &arch);
   ^~~~~~~~~~~~~~~~~~
   image_get_arch
common/image-fit.c:423:3: warning: implicit declaration of function 'fit_image_get_os'; did you mean 'image_get_os'? [-Wimplicit-function-declaration]
   fit_image_get_os(fit, image_noffset, &os);
   ^~~~~~~~~~~~~~~~
   image_get_os
common/image-fit.c:430:9: warning: implicit declaration of function 'fit_image_get_load'; did you mean 'image_get_load'? [-Wimplicit-function-declaration]
   ret = fit_image_get_load(fit, image_noffset, &load);
         ^~~~~~~~~~~~~~~~~~
         image_get_load
common/image-fit.c:444:9: warning: implicit declaration of function 'fit_image_get_entry'; did you mean 'fit_image_print'? [-Wimplicit-function-declaration]
   ret = fit_image_get_entry(fit, image_noffset, &entry);
         ^~~~~~~~~~~~~~~~~~~
         fit_image_print
common/image-fit.c: In function 'fit_get_timestamp':
common/image-fit.c:511:7: warning: assignment makes pointer from integer without a cast [-Wint-conversion]
  data = fdt_getprop(fit, noffset, FIT_TIMESTAMP_PROP, &len);
       ^
In file included from include/linux/bug.h:7:0,
                 from include/common.h:26,
                 from common/image-fit.c:18:
common/image-fit.c: In function 'fit_image_get_node':
common/image-fit.c:544:9: warning: format '%s' expects argument of type 'char *', but argument 3 has type 'int' [-Wformat=]
   debug("Can't find images parent node '%s' (%s)\n",
         ^
include/linux/printk.h:37:21: note: in definition of macro 'pr_fmt'
 #define pr_fmt(fmt) fmt
                     ^~~
include/log.h:142:2: note: in expansion of macro 'debug_cond'
  debug_cond(_DEBUG, fmt, ##args)
  ^~~~~~~~~~
common/image-fit.c:544:3: note: in expansion of macro 'debug'
   debug("Can't find images parent node '%s' (%s)\n",
   ^~~~~
common/image-fit.c:549:12: warning: implicit declaration of function 'fdt_subnode_offset' [-Wimplicit-function-declaration]
  noffset = fdt_subnode_offset(fit, images_noffset, image_uname);
            ^~~~~~~~~~~~~~~~~~
In file included from include/linux/bug.h:7:0,
                 from include/common.h:26,
                 from common/image-fit.c:18:
common/image-fit.c:551:9: warning: format '%s' expects argument of type 'char *', but argument 3 has type 'int' [-Wformat=]
   debug("Can't get node offset for image unit name: '%s' (%s)\n",
         ^
include/linux/printk.h:37:21: note: in definition of macro 'pr_fmt'
 #define pr_fmt(fmt) fmt
                     ^~~
include/log.h:142:2: note: in expansion of macro 'debug_cond'
  debug_cond(_DEBUG, fmt, ##args)
  ^~~~~~~~~~
common/image-fit.c:551:3: note: in expansion of macro 'debug'
   debug("Can't get node offset for image unit name: '%s' (%s)\n",
   ^~~~~
common/image-fit.c: In function 'fit_image_get_os':
common/image-fit.c:578:7: warning: assignment makes pointer from integer without a cast [-Wint-conversion]
  data = fdt_getprop(fit, noffset, FIT_OS_PROP, &len);
       ^
common/image-fit.c: In function 'fit_image_get_arch':
common/image-fit.c:610:7: warning: assignment makes pointer from integer without a cast [-Wint-conversion]
  data = fdt_getprop(fit, noffset, FIT_ARCH_PROP, &len);
       ^
common/image-fit.c: In function 'fit_image_get_type':
common/image-fit.c:642:7: warning: assignment makes pointer from integer without a cast [-Wint-conversion]
  data = fdt_getprop(fit, noffset, FIT_TYPE_PROP, &len);
       ^
common/image-fit.c: In function 'fit_image_get_comp':
common/image-fit.c:674:7: warning: assignment makes pointer from integer without a cast [-Wint-conversion]
  data = fdt_getprop(fit, noffset, FIT_COMP_PROP, &len);
       ^
common/image-fit.c: In function 'fit_image_get_address':
common/image-fit.c:690:8: error: unknown type name 'fdt32_t'
  const fdt32_t *cell;
        ^~~~~~~
common/image-fit.c:693:7: warning: assignment makes pointer from integer without a cast [-Wint-conversion]
  cell = fdt_getprop(fit, noffset, name, &len);
       ^
common/image-fit.c: In function 'fit_image_get_data':
common/image-fit.c:774:8: warning: assignment makes pointer from integer without a cast [-Wint-conversion]
  *data = fdt_getprop(fit, noffset, FIT_DATA_PROP, &len);
        ^
common/image-fit.c: In function 'fit_image_get_data_offset':
common/image-fit.c:798:8: error: unknown type name 'fdt32_t'
  const fdt32_t *val;
        ^~~~~~~
common/image-fit.c:800:6: warning: assignment makes pointer from integer without a cast [-Wint-conversion]
  val = fdt_getprop(fit, noffset, FIT_DATA_OFFSET_PROP, NULL);
      ^
common/image-fit.c:804:17: warning: implicit declaration of function 'fdt32_to_cpu'; did you mean 'be32_to_cpu'? [-Wimplicit-function-declaration]
  *data_offset = fdt32_to_cpu(*val);
                 ^~~~~~~~~~~~
                 be32_to_cpu
common/image-fit.c: In function 'fit_image_get_data_position':
common/image-fit.c:823:8: error: unknown type name 'fdt32_t'
  const fdt32_t *val;
        ^~~~~~~
common/image-fit.c:825:6: warning: assignment makes pointer from integer without a cast [-Wint-conversion]
  val = fdt_getprop(fit, noffset, FIT_DATA_POSITION_PROP, NULL);
      ^
common/image-fit.c: In function 'fit_image_get_data_size':
common/image-fit.c:847:8: error: unknown type name 'fdt32_t'
  const fdt32_t *val;
        ^~~~~~~
common/image-fit.c:849:6: warning: assignment makes pointer from integer without a cast [-Wint-conversion]
  val = fdt_getprop(fit, noffset, FIT_DATA_SIZE_PROP, NULL);
      ^
common/image-fit.c: In function 'fit_get_end':
common/image-fit.c:945:38: warning: implicit declaration of function 'fdt_totalsize' [-Wimplicit-function-declaration]
  return map_to_sysmem((void *)(fit + fdt_totalsize(fit)));
                                      ^~~~~~~~~~~~~
common/image-fit.c: In function 'fit_set_timestamp':
common/image-fit.c:967:8: warning: implicit declaration of function 'fdt_setprop'; did you mean 'fdt_getprop_u32'? [-Wimplicit-function-declaration]
  ret = fdt_setprop(fit, noffset, FIT_TIMESTAMP_PROP, &t,
        ^~~~~~~~~~~
        fdt_getprop_u32
In file included from include/linux/bug.h:7:0,
                 from include/common.h:26,
                 from common/image-fit.c:18:
common/image-fit.c:970:9: warning: format '%s' expects argument of type 'char *', but argument 3 has type 'int' [-Wformat=]
   debug("Can't set '%s' property for '%s' node (%s)\n",
         ^
include/linux/printk.h:37:21: note: in definition of macro 'pr_fmt'
 #define pr_fmt(fmt) fmt
                     ^~~
include/log.h:142:2: note: in expansion of macro 'debug_cond'
  debug_cond(_DEBUG, fmt, ##args)
  ^~~~~~~~~~
common/image-fit.c:970:3: note: in expansion of macro 'debug'
   debug("Can't set '%s' property for '%s' node (%s)\n",
   ^~~~~
common/image-fit.c:970:9: warning: format '%s' expects argument of type 'char *', but argument 4 has type 'int' [-Wformat=]
   debug("Can't set '%s' property for '%s' node (%s)\n",
         ^
include/linux/printk.h:37:21: note: in definition of macro 'pr_fmt'
 #define pr_fmt(fmt) fmt
                     ^~~
include/log.h:142:2: note: in expansion of macro 'debug_cond'
  debug_cond(_DEBUG, fmt, ##args)
  ^~~~~~~~~~
common/image-fit.c:970:3: note: in expansion of macro 'debug'
   debug("Can't set '%s' property for '%s' node (%s)\n",
   ^~~~~
common/image-fit.c:973:18: error: 'FDT_ERR_NOSPACE' undeclared (first use in this function); did you mean 'FDT_ERROR'?
   return ret == -FDT_ERR_NOSPACE ? -ENOSPC : -1;
                  ^~~~~~~~~~~~~~~
                  FDT_ERROR
common/image-fit.c:973:18: note: each undeclared identifier is reported only once for each function it appears in
common/image-fit.c: In function 'calculate_hash':
common/image-fit.c:1000:6: error: 'IMAGE_ENABLE_CRC32' undeclared (first use in this function); did you mean 'IMAGE_ENABLE_FIT'?
  if (IMAGE_ENABLE_CRC32 && strcmp(algo, "crc32") == 0) {
      ^~~~~~~~~~~~~~~~~~
      IMAGE_ENABLE_FIT
common/image-fit.c:1005:13: error: 'IMAGE_ENABLE_SHA1' undeclared (first use in this function); did you mean 'IMAGE_ENABLE_FIT'?
  } else if (IMAGE_ENABLE_SHA1 && strcmp(algo, "sha1") == 0) {
             ^~~~~~~~~~~~~~~~~
             IMAGE_ENABLE_FIT
common/image-fit.c:1009:13: error: 'IMAGE_ENABLE_SHA256' undeclared (first use in this function); did you mean 'IMAGE_ENABLE_SHA1'?
  } else if (IMAGE_ENABLE_SHA256 && strcmp(algo, "sha256") == 0) {
             ^~~~~~~~~~~~~~~~~~~
             IMAGE_ENABLE_SHA1
common/image-fit.c:1013:13: error: 'IMAGE_ENABLE_MD5' undeclared (first use in this function); did you mean 'IMAGE_ENABLE_FIT'?
  } else if (IMAGE_ENABLE_MD5 && strcmp(algo, "md5") == 0) {
             ^~~~~~~~~~~~~~~~
             IMAGE_ENABLE_FIT
In file included from include/common.h:39:0,
                 from common/image-fit.c:18:
common/image-fit.c: In function 'fit_image_check_hash':
include/image.h:921:26: error: 'HASH_MAX_DIGEST_SIZE' undeclared (first use in this function)
 #define FIT_MAX_HASH_LEN HASH_MAX_DIGEST_SIZE
                          ^
common/image-fit.c:1026:16: note: in expansion of macro 'FIT_MAX_HASH_LEN'
  uint8_t value[FIT_MAX_HASH_LEN];
                ^~~~~~~~~~~~~~~~
common/image-fit.c:1026:10: warning: unused variable 'value' [-Wunused-variable]
  uint8_t value[FIT_MAX_HASH_LEN];
          ^~~~~
common/image-fit.c: In function 'fit_image_verify':
common/image-fit.c:1100:6: error: 'IMAGE_ENABLE_VERIFY' undeclared (first use in this function); did you mean 'IMAGE_ENABLE_FIT'?
  if (IMAGE_ENABLE_VERIFY &&
      ^~~~~~~~~~~~~~~~~~~
      IMAGE_ENABLE_FIT
common/image-fit.c:1101:6: warning: implicit declaration of function 'fit_image_verify_required_sigs'; did you mean 'fit_image_verify'? [-Wimplicit-function-declaration]
      fit_image_verify_required_sigs(fit, image_noffset, data, size,
      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      fit_image_verify
common/image-fit.c:1102:9: warning: implicit declaration of function 'gd_fdt_blob' [-Wimplicit-function-declaration]
         gd_fdt_blob(), &verify_all)) {
         ^~~~~~~~~~~
common/image-fit.c:1108:2: warning: implicit declaration of function 'fdt_for_each_subnode'; did you mean 'list_for_each_safe'? [-Wimplicit-function-declaration]
  fdt_for_each_subnode(noffset, fit, image_noffset) {
  ^~~~~~~~~~~~~~~~~~~~
  list_for_each_safe
common/image-fit.c:1108:52: error: expected ';' before '{' token
  fdt_for_each_subnode(noffset, fit, image_noffset) {
                                                    ^
common/image-fit.c:1149:28: warning: format '%s' expects argument of type 'char *', but argument 3 has type 'int' [-Wformat=]
  printf(" error!\n%s for '%s' hash node in '%s' image node\n",
                           ~^
                           %d
         err_msg, fit_get_name(fit, noffset, NULL),
                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
common/image-fit.c:1149:46: warning: format '%s' expects argument of type 'char *', but argument 4 has type 'int' [-Wformat=]
  printf(" error!\n%s for '%s' hash node in '%s' image node\n",
                                             ~^
                                             %d
common/image-fit.c:1151:9:
         fit_get_name(fit, image_noffset, NULL));
         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
common/image-fit.c:1091:6: warning: unused variable 'ret' [-Wunused-variable]
  int ret;
      ^~~
common/image-fit.c: In function 'fit_all_image_verify':
common/image-fit.c:1176:48: warning: format '%s' expects argument of type 'char *', but argument 3 has type 'int' [-Wformat=]
   printf("Can't find images parent node '%s' (%s)\n",
                                               ~^
                                               %d
          FIT_IMAGES_PATH, fdt_strerror(images_noffset));
                           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
common/image-fit.c:1193:39: warning: format '%s' expects argument of type 'char *', but argument 3 has type 'int' [-Wformat=]
    printf("   Hash(es) for Image %u (%s): ", count,
                                      ~^
                                      %d
           fit_get_name(fit, noffset, NULL));
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
common/image-fit.c: In function 'fit_check_format':
common/image-fit.c:1316:47: warning: comparison between pointer and integer
  if (fdt_getprop(fit, 0, FIT_DESC_PROP, NULL) == NULL) {
                                               ^~
common/image-fit.c:1323:53: warning: comparison between pointer and integer
   if (fdt_getprop(fit, 0, FIT_TIMESTAMP_PROP, NULL) == NULL) {
                                                     ^~
common/image-fit.c: In function 'fit_conf_find_compat':
common/image-fit.c:1394:13: warning: assignment makes pointer from integer without a cast [-Wint-conversion]
  fdt_compat = fdt_getprop(fdt, 0, "compatible", &fdt_compat_len);
             ^
common/image-fit.c:1417:13: warning: assignment makes pointer from integer without a cast [-Wint-conversion]
   kfdt_name = fdt_getprop(fit, noffset, "fdt", &len);
             ^
common/image-fit.c:1447:9: warning: implicit declaration of function 'fdt_node_check_compatible'; did you mean 'fit_image_check_comp'? [-Wimplicit-function-declaration]
    if (!fdt_node_check_compatible(kfdt, 0,
         ^~~~~~~~~~~~~~~~~~~~~~~~~
         fit_image_check_comp
In file included from include/linux/bug.h:7:0,
                 from include/common.h:26,
                 from common/image-fit.c:18:
common/image-fit.c: In function 'fit_conf_get_node':
common/image-fit.c:1492:9: warning: format '%s' expects argument of type 'char *', but argument 3 has type 'int' [-Wformat=]
   debug("Can't find configurations parent node '%s' (%s)\n",
         ^
include/linux/printk.h:37:21: note: in definition of macro 'pr_fmt'
 #define pr_fmt(fmt) fmt
                     ^~~
include/log.h:142:2: note: in expansion of macro 'debug_cond'
  debug_cond(_DEBUG, fmt, ##args)
  ^~~~~~~~~~
common/image-fit.c:1492:3: note: in expansion of macro 'debug'
   debug("Can't find configurations parent node '%s' (%s)\n",
   ^~~~~
common/image-fit.c:1526:9: warning: format '%s' expects argument of type 'char *', but argument 3 has type 'int' [-Wformat=]
   debug("Can't get node offset for configuration unit name: '%s' (%s)\n",
         ^
include/linux/printk.h:37:21: note: in definition of macro 'pr_fmt'
 #define pr_fmt(fmt) fmt
                     ^~~
include/log.h:142:2: note: in expansion of macro 'debug_cond'
  debug_cond(_DEBUG, fmt, ##args)
  ^~~~~~~~~~
common/image-fit.c:1526:3: note: in expansion of macro 'debug'
   debug("Can't get node offset for configuration unit name: '%s' (%s)\n",
   ^~~~~
common/image-fit.c: In function 'fit_conf_get_prop_node_count':
common/image-fit.c:1539:9: warning: implicit declaration of function 'fdt_stringlist_count' [-Wimplicit-function-declaration]
  return fdt_stringlist_count(fit, noffset, prop_name);
         ^~~~~~~~~~~~~~~~~~~~
common/image-fit.c: In function 'fit_conf_get_prop_node_index':
common/image-fit.c:1549:10: warning: implicit declaration of function 'fdt_stringlist_get'; did you mean '__stringify_1'? [-Wimplicit-function-declaration]
  uname = fdt_stringlist_get(fit, noffset, prop_name, index, &len);
          ^~~~~~~~~~~~~~~~~~
          __stringify_1
common/image-fit.c:1549:8: warning: assignment makes pointer from integer without a cast [-Wint-conversion]
  uname = fdt_stringlist_get(fit, noffset, prop_name, index, &len);
        ^
common/image-fit.c: At top level:
common/image-fit.c:1574:6: warning: conflicting types for 'fit_conf_print'
 void fit_conf_print(const void *fit, int noffset, const char *p)
      ^~~~~~~~~~~~~~
common/image-fit.c:243:4: note: previous implicit declaration of 'fit_conf_print' was here
    fit_conf_print(fit, noffset, p);
    ^~~~~~~~~~~~~~
common/image-fit.c: In function 'fit_conf_print':
common/image-fit.c:1589:8: warning: assignment makes pointer from integer without a cast [-Wint-conversion]
  uname = fdt_getprop(fit, noffset, FIT_KERNEL_PROP, NULL);
        ^
common/image-fit.c:1597:8: warning: assignment makes pointer from integer without a cast [-Wint-conversion]
  uname = fdt_getprop(fit, noffset, FIT_RAMDISK_PROP, NULL);
        ^
common/image-fit.c:1602:13: warning: assignment makes pointer from integer without a cast [-Wint-conversion]
       uname = fdt_stringlist_get(fit, noffset, FIT_FDT_PROP,
             ^
common/image-fit.c:1613:8: warning: assignment makes pointer from integer without a cast [-Wint-conversion]
  uname = fdt_getprop(fit, noffset, FIT_FPGA_PROP, NULL);
        ^
common/image-fit.c:1619:13: warning: assignment makes pointer from integer without a cast [-Wint-conversion]
       uname = fdt_stringlist_get(fit, noffset, FIT_LOADABLE_PROP,
             ^
In file included from include/common.h:48:0,
                 from common/image-fit.c:18:
common/image-fit.c: In function 'fit_get_node_from_config':
common/image-fit.c:1655:25: error: 'bootm_headers_t {aka struct bootm_headers}' has no member named 'fit_uname_cfg'
        prop_name, images->fit_uname_cfg, addr);
                         ^
include/log.h:135:26: note: in definition of macro 'debug_cond'
    printf(pr_fmt(fmt), ##args); \
                          ^~~~
common/image-fit.c:1654:2: note: in expansion of macro 'debug'
  debug("*  %s: using config '%s' from image at 0x%08lx\n",
  ^~~~~
common/image-fit.c:1659:49: error: 'bootm_headers_t {aka struct bootm_headers}' has no member named 'fit_uname_cfg'
  cfg_noffset = fit_conf_get_node(fit_hdr, images->fit_uname_cfg);
                                                 ^~
common/image-fit.c: In function 'fit_image_load':
common/image-fit.c:1751:7: error: 'IMAGE_ENABLE_BEST_MATCH' undeclared (first use in this function); did you mean 'IMAGE_ENABLE_TIMESTAMP'?
   if (IMAGE_ENABLE_BEST_MATCH && !fit_uname_config) {
       ^~~~~~~~~~~~~~~~~~~~~~~
       IMAGE_ENABLE_TIMESTAMP
common/image-fit.c:1752:44: warning: passing argument 2 of 'fit_conf_find_compat' makes pointer from integer without a cast [-Wint-conversion]
    cfg_noffset = fit_conf_find_compat(fit, gd_fdt_blob());
                                            ^~~~~~~~~~~
common/image-fit.c:1378:5: note: expected 'const void *' but argument is of type 'int'
 int fit_conf_find_compat(const void *fit, const void *fdt)
     ^~~~~~~~~~~~~~~~~~~~
common/image-fit.c:1763:27: warning: implicit declaration of function 'fdt_get_name'; did you mean 'eth_get_name'? [-Wimplicit-function-declaration]
   fit_base_uname_config = fdt_get_name(fit, cfg_noffset, NULL);
                           ^~~~~~~~~~~~
                           eth_get_name
common/image-fit.c:1763:25: warning: assignment makes pointer from integer without a cast [-Wint-conversion]
   fit_base_uname_config = fdt_get_name(fit, cfg_noffset, NULL);
                         ^
common/image-fit.c:1767:10: error: 'bootm_headers_t {aka struct bootm_headers}' has no member named 'fit_uname_cfg'
    images->fit_uname_cfg = fit_base_uname_config;
          ^~
common/image-fit.c:1768:8: error: 'IMAGE_ENABLE_VERIFY' undeclared (first use in this function); did you mean 'IMAGE_ENABLE_FIT'?
    if (IMAGE_ENABLE_VERIFY && images->verify) {
        ^~~~~~~~~~~~~~~~~~~
        IMAGE_ENABLE_FIT
common/image-fit.c:1770:9: warning: implicit declaration of function 'fit_config_verify'; did you mean 'fit_conf_print'? [-Wimplicit-function-declaration]
     if (fit_config_verify(fit, cfg_noffset)) {
         ^~~~~~~~~~~~~~~~~
         fit_conf_print
common/image-fit.c:1783:13: warning: assignment makes pointer from integer without a cast [-Wint-conversion]
   fit_uname = fit_get_name(fit, noffset, NULL);
             ^
common/image-fit.c:1801:7: warning: implicit declaration of function 'fit_image_check_target_arch'; did you mean 'image_check_target_arch'? [-Wimplicit-function-declaration]
  if (!fit_image_check_target_arch(fit, noffset)) {
       ^~~~~~~~~~~~~~~~~~~~~~~~~~~
       image_check_target_arch
common/image-fit.c:1863:38: warning: implicit declaration of function 'fdt_check_header'; did you mean 'fit_check_format'? [-Wimplicit-function-declaration]
  if (image_type == IH_TYPE_FLATDT && fdt_check_header(buf)) {
                                      ^~~~~~~~~~~~~~~~
                                      fit_check_format
common/image-fit.c:1900:22: warning: implicit declaration of function 'fit_get_size'; did you mean 'fit_get_end'? [-Wimplicit-function-declaration]
   image_end = addr + fit_get_size(fit);
                      ^~~~~~~~~~~~
                      fit_get_end
common/image-fit.c: In function 'boot_get_setup_fit':
common/image-fit.c:1937:31: error: 'bootm_headers_t {aka struct bootm_headers}' has no member named 'fit_hdr_os'; did you mean 'legacy_hdr_os'?
  addr = map_to_sysmem(images->fit_hdr_os);
                               ^~~~~~~~~~
                               legacy_hdr_os
At top level:
common/image-fit.c:1023:12: warning: 'fit_image_check_hash' defined but not used [-Wunused-function]
 static int fit_image_check_hash(const void *fit, int noffset, const void *data,
            ^~~~~~~~~~~~~~~~~~~~
scripts/Makefile.build:280: recipe for target 'common/image-fit.o' failed
make[4]: *** [common/image-fit.o] Error 1
Makefile:1330: recipe for target 'common' failed
make[3]: *** [common] Error 2
make[3]: Leaving directory '/home/hauke/openwrt/lede/build_dir/target-arm_cortex-a9+neon_musl_eabi/u-boot-wandboard/u-boot-2018.03'
Makefile:66: recipe for target '/home/hauke/openwrt/lede/build_dir/target-arm_cortex-a9+neon_musl_eabi/u-boot-wandboard/u-boot-2018.03/.built' failed
make[2]: *** [/home/hauke/openwrt/lede/build_dir/target-arm_cortex-a9+neon_musl_eabi/u-boot-wandboard/u-boot-2018.03/.built] Error 2
make[2]: Leaving directory '/home/hauke/openwrt/lede/package/boot/uboot-imx6'
Command exited with non-zero status 2
time: package/boot/uboot-imx6/wandboard/compile#18.88#1.64#22.69
package/Makefile:107: recipe for target 'package/boot/uboot-imx6/compile' failed
make[1]: *** [package/boot/uboot-imx6/compile] Error 2
make[1]: Leaving directory '/home/hauke/openwrt/lede'
/home/hauke/openwrt/lede/include/toplevel.mk:216: recipe for target 'package/uboot-imx6/compile' failed
make: *** [package/uboot-imx6/compile] Error 2
```

When I remove the patches I get this error message:
```
  CFG     u-boot.cfg
  CFGCHK  u-boot.cfg
make[3]: Leaving directory '/home/hauke/openwrt/lede/build_dir/target-arm_cortex-a9+neon_musl_eabi/u-boot-wandboard/u-boot-2018.03'
touch /home/hauke/openwrt/lede/build_dir/target-arm_cortex-a9+neon_musl_eabi/u-boot-wandboard/u-boot-2018.03/.built
rm -rf /home/hauke/openwrt/lede/build_dir/target-arm_cortex-a9+neon_musl_eabi/u-boot-wandboard/u-boot-2018.03/.pkgdir/u-boot-wandboard /home/hauke/openwrt/lede/build_dir/target-arm_cortex-a9+neon_musl_eabi/u-boot-wandboard/u-boot-2018.03/.pkgdir/u-boot-wandboard.installed
mkdir -p /home/hauke/openwrt/lede/build_dir/target-arm_cortex-a9+neon_musl_eabi/u-boot-wandboard/u-boot-2018.03/.pkgdir/u-boot-wandboard
cp -fpR /home/hauke/openwrt/lede/build_dir/target-arm_cortex-a9+neon_musl_eabi/u-boot-wandboard/u-boot-2018.03/u-boot.imx /home/hauke/openwrt/lede/build_dir/target-arm_cortex-a9+neon_musl_eabi/u-boot-wandboard/u-boot-2018.03/.pkgdir/u-boot-wandboard/
cp: cannot stat '/home/hauke/openwrt/lede/build_dir/target-arm_cortex-a9+neon_musl_eabi/u-boot-wandboard/u-boot-2018.03/u-boot.imx': No such file or directory
Makefile:66: recipe for target '/home/hauke/openwrt/lede/build_dir/target-arm_cortex-a9+neon_musl_eabi/u-boot-wandboard/u-boot-2018.03/.pkgdir/u-boot-wandboard.installed' failed
make[2]: *** [/home/hauke/openwrt/lede/build_dir/target-arm_cortex-a9+neon_musl_eabi/u-boot-wandboard/u-boot-2018.03/.pkgdir/u-boot-wandboard.installed] Error 1
make[2]: Leaving directory '/home/hauke/openwrt/lede/package/boot/uboot-imx6'
Command exited with non-zero status 2
time: package/boot/uboot-imx6/wandboard/compile#42.86#3.64#51.03
package/Makefile:107: recipe for target 'package/boot/uboot-imx6/compile' failed
make[1]: *** [package/boot/uboot-imx6/compile] Error 2
make[1]: Leaving directory '/home/hauke/openwrt/lede'
/home/hauke/openwrt/lede/include/toplevel.mk:216: recipe for target 'package/uboot-imx6/compile' failed
make: *** [package/uboot-imx6/compile] Error 2
```

v-vid commented:
Hi @hauke 
I'm not entirely sure what was changed to broke the build, but second commit (https://github.com/openwrt/openwrt/pull/904/commits/4c3c40ad1ae97f9f0182772d490432151dec761e) resolves 2 newly introduced issues:

1. FIT related options are moved from wandboard.h to wanboard_defconfig to resolve declaration issues
2. u-boot.imx is now explicitly defined in Build/Compile section, otherwise all target will fail with "Cannot find u-boot.imx"

If somebody could test this as well and provide feedback, it would be much appreciated.

hauke commented on package/boot/uboot-imx6/Makefile:
```diff
@@ -48,31 +48,24 @@ define U-Boot/nitrogen6s1g
+define Build/Compile
+	$(MAKE) -C $(PKG_BUILD_DIR) \
+		CROSS_COMPILE=$(TARGET_CROSS) u-boot.imx
+endef
```
Instead of manually defining the compile set please just define that you want to have the u-boot.imx image like this:
UBOOT_MAKE_FLAGS += u-boot.imx

hauke commented on package/boot/uboot-imx6/Makefile:
```diff
@@ -7,10 +7,10 @@
 PKG_RELEASE:=1
 
-PKG_HASH:=7b6444bd23eb61068c43bd1d44ec7e7bfdbce5cadeca20c833eee186b4d3fd31
+PKG_HASH:=02922bdf0ee003fe25bfc32749ffdeab
```
Please use a SHA256 hash this looks too small.

hauke reviewed (changes requested):
Please merge these two commits into one as these changes are all needed to make this build.
Was this was not even build tested in the beginning, what tests have you done in addition?

v-vid commented:
Hi @hauke,

merged commits, added sha256 hash instead of md5 and replaced "Compile" section with UBOOT_MAKE_FLAGS. Re-tested everything again.
Yes, all targets were tested before submitting PR and @tmn505 also commented above that the build was fine.

hauke commented:
Thank you for the patch, it was applied to master.
